lib/pages/chat/chat_input_row.dart

```diff
@@ -214,6 +127,178 @@ class ChatInputRow extends StatelessWidget {
+        color: Theme.of(context).colorScheme.surface,
+      ),
+      child: Row(
+        crossAxisAlignment: CrossAxisAlignment.end,
+        children: [
+          TwakeIconButton(
+            tooltip: L10n.of(context)!.more,
+            paddingAll: 0.0,
+            margin: const EdgeInsets.all(12.0),
+            icon: Icons.add_circle_outline,
+            size: 24.0,
+            onTap: () => controller.onSendFileClick(context),
+          ),
+          Expanded(
+            child: _buildInputBar(context),
+          ),
+          KeyBoardShortcuts(
+            keysToPress: {LogicalKeyboardKey.altLeft, LogicalKeyboardKey.keyE},
+            onKeysPressed: controller.emojiPickerAction,
+            helpLabel: L10n.of(context)!.emojis,
+            child: InkWell(
+              onTap: controller.emojiPickerAction,
+              hoverColor: Colors.transparent,
+              child: PageTransitionSwitcher(
+                transitionBuilder: (
+                  Widget child,
+                  Animation<double> primaryAnimation,
+                  Animation<double> secondaryAnimation,
+                ) {
+                  return SharedAxisTransition(
+                    animation: primaryAnimation,
+                    secondaryAnimation: secondaryAnimation,
+                    transitionType: SharedAxisTransitionType.scaled,
+                    fillColor: Colors.transparent,
+                    child: child,
+                  );
+                },
+                child: !controller.showEmojiPicker
+                    ? ValueListenableBuilder(
+                        valueListenable: controller.inputText,
+                        builder: (context, value, child) {
+                          return TwakeIconButton(
+                            margin: const EdgeInsets.all(12.0),
+                            paddingAll: 0.0,
+                            size: 24.0,
+                            tooltip: L10n.of(context)!.emojis,
+                            onTap: () => controller.emojiPickerAction(),
+                            icon: Icons.tag_faces,
+                          );
+                        },
+                      )
+                    : ValueListenableBuilder(
+                        valueListenable: controller.inputText,
+                        builder: (context, value, child) {
+                          return TwakeIconButton(
+                            margin: const EdgeInsets.all(12.0),
+                            paddingAll: 0,
+                            size: 24.0,
+                            tooltip: L10n.of(context)!.keyboard,
+                            onTap: () => controller.inputFocus.requestFocus(),
+                            icon: Icons.keyboard,
+                          );
+                        },
+                      ),
+              ),
+            ),
+          ),
+        ],
+      ),
+    );
+  }
+
+  InputBar _buildInputBar(BuildContext context) {
```

Comment:
same

Reply:
I agree for the other 2 but why this one ?

Reply:
because it's also a stateless widget, so make it as much as possible, it will be easier when you debug.

Reply:
@nqhhdev @drminh2807 @hoangdat wdyt ? for me we'll create a widget with same arguments there is no big performance gain

Reply:
IMO: No need create a stl

Reply:
IMO, no need this case